config.c

```diff
@@ -59,7 +57,7 @@ config_t config_open(const char *fname) {
 		/*
 		 * Find first non-empty character
 		 */
-		for (i = j = 0; j < len && isspace((u_char)buf[j]); ++j);
+		for (j = 0; j < len && isspace((u_char)buf[j]); ++j);
```

Comment:
I was never sure if removing setting `i` here is safe (and making similar changes in other places). If the loop is continued before `i` is set otherwise, we may get some different behavior.
I would be very cautious with the warnings and analyzer "issues". As long as we do not have tests that verify correct behavior for different use cases it might be dangerous to touch such legacy code.

Reply:
I had to read this code a few times before realising that the analyser is correct: `i` is assigned at line 62 and then reassigned at line 73, it is not read in between. The `continue` at line 68 ends the current iteration, so it never happens that `i` is read in an undefined state.

The meaning of this warning (dead code) is exactly this: don't assign to a variable if the variable is not read, because it is confusing (that is, don't do double assignments).

There are other places in the codebase where these dead codes are even trickier to spot, but the analyser is always correct.

Reply:
Ok. I trust you with this :) I am absolutely pro analyzers, but I have also often enough seen that they report false positives (and false negatives).

Reply:
I fully agree with you. We must be very careful with analysers. Indeed `scan-build` also warns about this:
```
Compiling xcrypt.c
xcrypt.c:840:10: warning: Assigned value is garbage or undefined [core.uninitialized.Assign]
    *d++ ^= *s++;
         ^  ~~~~
1 warning generated.
```
This is a false positive, because `s` for sure points to a valid data (this warning can also happen when casting does not guarantee proper data alignment).

This warning happens in the github action that uses `scan-build: Using '/usr/lib/llvm-14/bin/clang' for static analysis`. I tested on a more recent version of ubuntu that uses llvm-16 (`scan-build: Using '/usr/lib/llvm-16/bin/clang' for static analysis`) and the warning is not present. 

For example, `Deepsource` that you recently added shows a lot of false positives. It will never succeed.